File: trunk/RL-Glue/Java/src/rlglue/agent/Agent.java
package rlglue.agent;

import rlglue.types.Action;
import rlglue.types.Observation;

public interface Agent
{
    public void agent_init(final String taskSpecification);
    public Action agent_start(Observation observation);
    public Action agent_step(double reward, Observation observation);
    public void agent_end(double reward);
    public void agent_cleanup();
    public void agent_freeze();
    public String agent_message(final String message);
}

File: RL-Glue/Python/rlglue/agent/Agent.py
from rlglue.types import Action
from rlglue.types import Observation

class Agent:
	# (string) -> void
	def agent_init(taskSpecification):
		pass
	
	# (Observation) -> Action
	def agent_start(observation):
		pass
	
	# (double, Observation) -> Action
	def agent_step(reward, observation):
		pass
	
	# (double) -> void
	def agent_end(reward):
		pass
	
	# () -> void
	def agent_cleanup():
		pass
	
	# () -> void
	def agent_freeze():
		pass
	
	# (string) -> string
	def agent_message(message):
		pass
